fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/LAONetworkManager.java

```diff
@@ -134,12 +136,9 @@ public Completable subscribe(Channel channel) {
         // This is used when reconnecting after a lost connection
         .doOnSuccess(answer -> subscribedChannels.add(channel))
         // Catchup already sent messages after the subscription to the channel is complete
-        // || TODO This should be used instead of the two next uncommented lines as it allows the
-        // || returned Completable to complete only when both subscribe and catchup are complete.
-        // || But right now, the LAO creation need this specific behavior.
-        // .flatMapCompletable(answer -> catchup(channel))
-        .doAfterSuccess(answer -> catchup(channel).subscribe())
-        .ignoreElement();
+        // This allows for the completion of the returned completable only when both subscribe
+        // and catchup are completed
+        .flatMapCompletable(answer -> catchup(channel));
```

Comment:
The reason we used `doAfterSuccess()` is to catchup after the subscribe is finished. Because in the current app implementation, the Lao needs to be created before message are retrieved through the catchup. (which is a very questionable design choice 😅)

This change does fix a problem, but it creates another one that can be very tricky to debug as it introduce concurrency.